dough-protection/src/main/java/io/github/bakedlibs/dough/protection/modules/RedProtectProtectionModule.java

```diff
@@ -46,7 +49,15 @@ public boolean hasPermission(OfflinePlayer p, Location l, Interaction action) {
         Player player = (Player) p;
         switch (action) {
             case INTERACT_BLOCK:
-                return region.canChest(player);
+                if (l.getBlock() instanceof Container){
+                    return region.canChest(player);
+                }
+                if (l.getBlock().getBlockData() instanceof Powerable){
+                    return region.canButton(player) || region.canLever(player);
+                }
+                if (l.getBlock().getBlockData() instanceof Openable){
+                    return region.canDoor(player);
+                }
```

Comment:
`Block#getBlockData()` is a rather expensive check, `Block#getState()` is even worse.
A material check would yield significantly better performance here

Also, you are potentially calling `Block#getBlockData()` twice per run here too which is redundant.